### ISSUE TITLE
Job keeping retry: `Yarn container exit handler: EXIT signal received in yarn container, performing clean up action...`

### ISSUE THREAD

**Issue by hao1939** [Microsoft/pai#1793]
Please help to figure out the cause of retry.

http://*.239/view.html?username=core&jobName=debug-005-one-gpu
![image](https://user-images.githubusercontent.com/1547343/49209137-a627dc00-f3f4-11e8-91dc-f369fafa6531.png)

//changing the ip to a dummy value

**Comment by mzmssg** [comment]
From the log, it seems that docker container is killed for some reason.
http://52.247.220.239/yarn/10.0.0.4:8188/applicationhistory/logs/10.0.0.14:8041/container_1542939248333_0648_01_000002/container_1542939248333_0648_01_000002/core
```
+[09:19:42] echo '[DEBUG] Yarn container exit handler: trying to kill docker container core-debug-005-one-gpu-container_1542939248333_0648_01_000002'
++[09:19:42] docker inspect '--format={{.State.Pid}}' core-debug-005-one-gpu-container_1542939248333_0648_01_000002
+[09:19:42] pid=
+[09:19:42] '[' ']'
+[09:19:42] debug_log 'Yarn container exit handler' 'docker container core-debug-005-one-gpu-container_1542939248333_0648_01_000002 has already exited'
```

I think the most possibility is the container consumes exceed resource,  then killed by os.

**Comment by hao1939** [comment]
@Gerhut , maybe you could test your fix with this job, it's from the end user.

**Comment by hao1939** [comment]
Not so easy.

We run job container with the `-rm` option, so we can't get the status using `docker inspect` after it quit.
Those logs come from system log.
@Gerhut was trying to collect such info from system log on the node, but the PR hadn't merged.

**Comment by Gerhut** [comment]
Currently, it is the docker container itself which will be killed, exit code is 137 (SIGKILL), `.State.OOMKilled` is `true`, but it cannot be found that "cgroup out of memory" in dmesg. Will do more investigations.

**Comment by Gerhut** [comment]
It is confirmed that the `python3` process is killed by system while `.State.Pid` yields pid of `bash`, which is the parent process of `python3`.

**Comment by scarlett2018** [comment]
Adding @qfyin for reference.

**Comment by scarlett2018** [comment]
In addition to how to expose logs for this issue, we should also consider reporting IT admin for this as well. But that has a lower priority than diagnosis-ability.

**Comment by fanyangCS** [comment]
@hao1939 , maybe we can remove the "-rm" flag and use a separate docker -rm command after collecting the log?

**Comment by hao1939** [comment]
So we will need to carefully design the container cleanup logical, make sure we get the info before it was cleaned.

**Comment by Gerhut** [comment]
Fixed by #1108
